Preston_Kellen/Assignments/login_registration/server.py

```diff
@@ -0,0 +1,128 @@
+from flask import Flask, render_template, request, redirect, session, flash
+from mysqlconnection import MySQLConnector
+from flask_bcrypt import Bcrypt
+import re
+EMAIL_REGEX = re.compile(r'^[a-zA-Z0-9.+_-]+@[a-zA-Z0-9._-]+\.[a-zA-Z]+$')
+
+# Below is how we will utilize md5
+# password = 'password';
+# encrypted_password = md5.new(password).hexdigest();
+
+app = Flask(__name__)
+app.secret_key = 'authenticate!@%#'
+bcrypt = Bcrypt(app)
+mysql = MySQLConnector(app,'wordwalldb')
+
+# valid_form runs all our 'new user' validation requirements
+def valid_logon():
+    if (len(request.form['username']) < 1 or
+    len(request.form['password']) < 1):
+        flash('Please enter your username and password', 'error')
+        return False
+    return True
+
+def valid_form():
+    if (len(request.form['first_name']) < 1 or
+    len(request.form['last_name']) < 1 or
+    len(request.form['last_name']) < 1 or
+    len(request.form['email']) < 1 or
+    len(request.form['password']) < 1 or
+    len(request.form['confpwd']) < 1 or
+    len(request.form['username']) < 1):
+        flash('Please fill out all forms', 'error')
+        return False
+    elif (not request.form['first_name'].isalpha() or not
+    request.form['last_name'].isalpha()):
+        flash('A name not a number', 'error')
+        return False
+    elif not EMAIL_REGEX.match(request.form['email']):
+        flash('Please enter a valid email address', 'error')
+        return False
+    elif len(request.form['password']) < 8:
+        flash('Password must be at least 8 characters', 'error')
+        return False
+    elif request.form['password'] != request.form['confpwd']:
+        flash('Passwords must match', 'error')
+        return False
+    query = "SELECT username FROM user"
+    # data = { 'username': username }
+    username_list = mysql.query_db(query)
+    for val in username_list:
```

Comment:
You want to use a WHERE statement in your query rather than pull every user in your DB.
Also, if you create your table setting email to UNIQUE, then you can use the try/catch block and forgo the unique check all together.